src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateCopy = addProperties(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        stateCopy = removeProperties(stateCopy, action.keysToRemove);
+        break;
+
+      case 'clear':
+        stateCopy = clearProperties();
```

Comment:
The `clearProperties` function is called without any arguments, but it is designed to return an empty object. Ensure that this behavior is intended and consistent with the task requirements. If `clearProperties` should reset `stateCopy`, the current implementation is correct. Otherwise, consider modifying the function to accept and clear the current state.